components/src/Button/Button.js

```diff
@@ -91,6 +91,7 @@ const Button = styled(BaseButton)(
     border: `1px solid ${theme.colors.darkBlue}`,
     borderRadius: theme.radii.medium,
     width: fullWidth ? "100%" : "auto",
+    margin: theme.space.none,
```

Comment:
Since space is above margin this will override any props passed in using styled-system margin. We probably want to allow margin changes for buttons as I can see some use cases for it. The easy fix would be to move space below the styles object here